src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,185 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+    private final RegistrationServiceImpl registrationServiceImpl = new RegistrationServiceImpl();
+    private StorageDao storageDao;
+
+    @BeforeEach
+    public void set_Up() {
+        storageDao = new StorageDaoImpl();
+    }
+
+    @Test
+    public void register_userNull_notOk() {
+        assertThrows(RegistrationException.class, () -> registrationServiceImpl.register(null),
+                "Invalid user date. User fields cannot be null");
+    }
+
+    @Test
+    public void register_validUser_ok() {
+        User validUser = new User();
+        validUser.setLogin("ValidUser");
+        validUser.setPassword("validPassword");
+        validUser.setAge(20);
+
+        registrationServiceImpl.register(validUser);
+        assertNotNull(storageDao.get("ValidUser"));
+
+        User firstUser = new User();
+        firstUser.setLogin("FirstUser");
+        firstUser.setPassword("firstPassword");
+        firstUser.setAge(20);
+
+        User secondUser = new User();
+        secondUser.setLogin("SecondUser");
+        secondUser.setPassword("secondPassword");
+        secondUser.setAge(25);
+
+        registrationServiceImpl.register(firstUser);
+        registrationServiceImpl.register(secondUser);
+
+        assertNotNull(storageDao.get("FirstUser"));
+        assertNotNull(storageDao.get("SecondUser"));
+
+        validUser.setLogin("UserLogin6");
+        assertTrue(Storage.people.add(validUser));
+
+        validUser.setLogin("UserLogin7");
+        assertTrue(Storage.people.add(validUser));
+
+        validUser.setLogin("UserLogin10");
+        assertTrue(Storage.people.add(validUser));
+
+        validUser.setPassword("abcdef");
+        assertTrue(Storage.people.add(validUser));
+
+        validUser.setPassword("abcdefg");
+        assertTrue(Storage.people.add(validUser));
+
+        validUser.setPassword("abcdefghij");
+        assertTrue(Storage.people.add(validUser));
+
+        validUser.setAge(18);
+        assertTrue(Storage.people.add(validUser));
+
+        validUser.setAge(19);
+        assertTrue(Storage.people.add(validUser));
+
+        validUser.setAge(25);
+        assertTrue(Storage.people.add(validUser));
+
+        validUser.setAge(60);
+        assertTrue(Storage.people.add(validUser));
+
+        validUser.setAge(Integer.MAX_VALUE);
+        assertTrue(Storage.people.add(validUser));
+    }
+
+    @Test
+    public void register_invalidLogin_notOk() {
+        User firstUser = new User();
```

Comment:
non informative naming of user, better name it like firstValidUser or firstInvalidUser, userInvalidAge etc